docs/design/ovs-pipeline.md

```diff
@@ -893,35 +922,42 @@ Flow 4 is the table-miss flow for case 3, matching packets from all new connecti
 ### PreRoutingClassifier
 
 This table handles the first packet from uncommitted Service connections before Service Endpoint selection. It
-sequentially resubmits the packets to tables [NodePortMark] and [SessionAffinity] to do some pre-processing, including
+sequentially resubmits the packets to tables [ServiceMark] and [SessionAffinity] to do some pre-processing, including
 the loading of specific reg marks. Subsequently, it forwards the packets to table [ServiceLB] to perform Service Endpoint
 selection.
 
 If you dump the flows of this table, you may see the following:
 
 ```text
-1. table=PreRoutingClassifier, priority=200,ip actions=resubmit(,NodePortMark),resubmit(,SessionAffinity),resubmit(,ServiceLB)
-2. table=PreRoutingClassifier, priority=0 actions=goto_table:NodePortMark
+1. table=PreRoutingClassifier, priority=200,ip actions=resubmit(,ServiceMark),resubmit(,SessionAffinity),resubmit(,ServiceLB)
+2. table=PreRoutingClassifier, priority=0 actions=goto_table:ServiceMark
 ```
 
-Flow 1 sequentially resubmits packets to tables [NodePortMark], [SessionAffinity], and [ServiceLB]. Note that packets
-are ultimately forwarded to table [ServiceLB]. In tables [NodePortMark] and [SessionAffinity], only reg marks are loaded.
+Flow 1 sequentially resubmits packets to tables [ServiceMark], [SessionAffinity], and [ServiceLB]. Note that packets
+are ultimately forwarded to table [ServiceLB]. In tables [ServiceMark] and [SessionAffinity], only reg marks are loaded.
 
 Flow 2 is the table-miss flow that should remain unused.
 
-### NodePortMark
+### ServiceMark
+
+This table is designed to mark Service traffic. It addresses specific cases:
 
-This table is designed to potentially mark packets destined for NodePort Services. It is only created when `proxyAll` is
-enabled.
+1. Packets potentially destined for NodePort Services.
+2. Packets destined for LoadBalancer Services configured with `loadBalancerSourceRanges`.
 
 If you dump the flows of this table, you may see the following:
 
 ```text
-1. table=NodePortMark, priority=200,ip,nw_dst=192.168.77.102 actions=set_field:0x80000/0x80000->reg4
-2. table=NodePortMark, priority=200,ip,nw_dst=169.254.0.252 actions=set_field:0x80000/0x80000->reg4
-3. table=NodePortMark, priority=0 actions=goto_table:SessionAffinity
+1. table=ServiceMark, priority=200,ip,nw_dst=192.168.77.102 actions=set_field:0x80000/0x80000->reg4
+2. table=ServiceMark, priority=200,ip,nw_dst=169.254.0.252 actions=set_field:0x80000/0x80000->reg4
+3. table=ServiceMark, priority=200,tcp,nw_src=192.168.77.0/24,nw_dst=192.168.77.152,tp_dst=80 actions=set_field:0x20000000/0x20000000->reg4",
+4. table=ServiceMark, priority=200,tcp,nw_src=192.168.78.0/24,nw_dst=192.168.77.152,tp_dst=80 actions=set_field:0x20000000/0x20000000->reg4",
+5. table=ServiceMark, priority=190,tcp,nw_dst=192.168.77.152,tp_dst=80 actions=set_field:0x4000/0x4000->reg0",
```

Comment:
I am probably missing something obvious, but what is the point of setting `LoadBalancerSourceRangesRegMark` in flows 3 and 4? If the source IP doesn't fall in the correct range, we mark the packet for rejection in flow 5, so it doesn't seem that we need to remember that the source IP was in the correct range?

Reply:
I have updated the commit message. Maybe it can help.